### DISCUSSION TITLE
Historic Development Firmwares for ESP32-C3

### DISCUSSION THREAD

**Discussion by 0allenc0** [arendst/Tasmota#21811] [Q&A]
I'm trying to find a firmware with bluetooth support for ESP32-C3 that doesn't require Safeboot (I may want to switch to ESPHome32 later) for a Switchbot Mini Plug - Flashed with Tasmota 11.1

I believe this requires ~~Development~~ unofficial ESP32-C3 firmware around version 12.5 or 13.2
Is there any place where I can get past ~~Development~~ unofficial OTA binaries / Firmwares?

**Comment by HofeBY** [comment]
https://web.archive.org/

To build your own Tasmota.
https://github.com/arendst/Tasmota/tree/prerelease-13.2

**Reply by 0allenc0** [reply]
I hadn't thought about that, but the archive only gives me a link and not the file (unless I'm missing something)

Development binaries for Tasmota firmware 13.1.0.4 on ESP32
https://web.archive.org/web/20231017205340/https://ota.tasmota.com/tasmota32/

**Reply by sfromis** [reply]
There are no "release" or "development" builds for ESP32-C3 with Bluetooth. That would be the "unofficial" ones, which are not archived very long time back. What you can do is to use the project source code (including older versions), and make a custom build. Beware that flashing a safeboot-less binary is not a single file, but you need the right setup with partition and bootloader files.

**Comment by Staars** [comment]
Did you look at:
https://jason2866.github.io/Tasmota-specials/
???

There is some (limited) archive. You probably want to try the BT-build.

**Comment by Jason2866** [comment]
Why not building an actual C3 firmware with BT small enough to OTA flash? BTW. idk why when you want to install esphome. Just install esphome now, as it is. If the device is flashed with the early 3rd party OTA converter you can't update the device via OTA.
